state/protocol/protocol_state/kvstore/models.go

```diff
@@ -190,9 +208,55 @@ func (model *Modelv1) GetProtocolStateVersion() uint64 {
 	return 1
 }
 
+// Modelv2 is v2 of the Protocol State key-value store.
+// This version adds the following changes:
+//   - Non-system-chunk service event validation support (adds ChunkBody.ServiceEventCount field)
+//   - EFM Recovery (adds EpochCommit.DKGIndexMap field)
+type Modelv2 struct {
```

Comment:
:thinking: This time we need to introduce a new model but for future upgrades I think it makes sense to separate protocol version from KV store version. If we had introduced an integer back in the day, we would have been able to perform this upgrade without cloning the module but simple upgrading one value. 

My proposal is to include a protocol version as separate field and use view based upgrader for it. This way we will be able to update protocol without making changes to the KV store schema or introducing new versions to the models. This is similar approach to what Alex was explaining in his last talk in working group for upgrading different parts of the software.

Reply:
After discussing this live, we decided to continue using the protocol state version to also stand in for behavioural-only changes.
- the overhead is relatively small
- we know we will need to introduce a distinct version for the execution stack, which will give us experience of dealing with multiple versions
- we will revisit this as we use the versioning system more going forward